dom.go

```diff
@@ -104,16 +111,25 @@ type RenderSkipper interface {
 type HTML struct {
 	node jsObject
 
-	namespace, tag, text, innerHTML string
-	styles, dataset                 map[string]string
-	properties, attributes          map[string]interface{}
-	eventListeners                  []*EventListener
-	children                        []ComponentOrHTML
+	namespace, tag, text, innerHTML, key string
+	styles, dataset                      map[string]string
+	properties, attributes               map[string]interface{}
+	eventListeners                       []*EventListener
+	children                             []ComponentOrHTML
+	keyMap                               map[string]ComponentOrHTML
+	nestedKeys                           bool
+	insertBeforeNode                     jsObject
+	lastRenderedChild                    *HTML
```

Comment:
The fields you've added here are not self-explanatory like the ones prior to this change. `key` is self explanatory, but:

- `keyMap`
- `nestedKeys`
- `insertBeforeNode`
- `lastRenderedChild`

are not. Can you document each of them with an explanation of what they are / what state they carry?